include/ddc/kernels/splines/splines_linear_problem_2x2_blocks.hpp

```diff
@@ -0,0 +1,258 @@
+// Copyright (C) The DDC development team, see COPYRIGHT.md file
+//
+// SPDX-License-Identifier: MIT
+
+#pragma once
+
+#include <cassert>
+#include <memory>
+#include <string>
+
+#include <Kokkos_DualView.hpp>
+
+#include "splines_linear_problem.hpp"
+#include "splines_linear_problem_dense.hpp"
+
+namespace ddc::detail {
+
+/**
+ * @brief A 2x2-blocks linear problem dedicated to the computation of a spline approximation (taking in account boundary conditions),
+ * with all blocks except top-left one being stored in dense format.
+ *
+ * A = |   Q    | gamma |
+ *     | lambda | delta |
```

Comment:
I think this is better for the documentation, otherwise the LU decomposition would be very hard to read with terms like `bottom_right - bottom_left*top_left^-1*top_right` in a matrix.